src/components/CalendarMonthGrid.jsx

```diff
@@ -84,29 +90,19 @@ export default class CalendarMonthGrid extends React.Component {
 
   componentWillReceiveProps(nextProps) {
     const { initialMonth, numberOfMonths } = nextProps;
-    const { months } = this.state;
 
     const hasMonthChanged = !this.props.initialMonth.isSame(initialMonth, 'month');
     const hasNumberOfMonthsChanged = this.props.numberOfMonths !== numberOfMonths;
-    let newMonths = months;
-
-    if (hasMonthChanged && !hasNumberOfMonthsChanged) {
-      if (initialMonth.isAfter(this.props.initialMonth)) {
-        newMonths = months.slice(1);
-        newMonths.push(months[months.length - 1].clone().add(1, 'month'));
-      } else {
-        newMonths = months.slice(0, months.length - 1);
-        newMonths.unshift(months[0].clone().subtract(1, 'month'));
-      }
-    }
 
-    if (hasNumberOfMonthsChanged) {
-      newMonths = getMonths(initialMonth, numberOfMonths);
+    if (hasMonthChanged || hasNumberOfMonthsChanged) {
```

Comment:
Why are you removing the performance gains from not creating a new moment object for every month on every transition? Like there was a reason these were separated and that change was v. recent.